pinot-tools/src/main/java/com/linkedin/pinot/tools/ClusterStateVerifier.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.tools;
+
+import com.linkedin.pinot.common.config.TableNameBuilder;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class ClusterStateVerifier {
+  private static final Logger LOGGER = LoggerFactory.getLogger(ClusterStateVerifier.class);
+  private PinotSegmentRebalancer _rebalancer;
```

Comment:
Let us not have this class depend on rebalancer. In fact, rebalance is now a controller command so you can remove the PinotSegmentRebalancer class altogether.

Reply:
That's a good idea! But let's separate these two tasks. I'll open another ticket to remove `PinotSegmentRebalancer` class.